### PR TITLE
Backlog automation for bz-2096414

### PR DESCRIPTION
**Pull request by DanielOsypenko**
New automation script for https://bugzilla.redhat.com/show_bug.cgi?id=2096414 test_odf_cephblockpool_compression_status

Validate Compression status for cephblockpool at StorageSystem details and ocs-storagecluster-cephblockpool are matching.
Some refactoring at ocs_ci/ocs/ui/validation_ui.py done in order to have more dry code

### CODE REVIEW

File: ocs_ci/ocs/ui/base_ui.py [hunk 1]
```diff
@@ -674,6 +676,82 @@ def navigate_block_pool_page(self):
         self.navigate_to_ocs_operator_page()
         self.do_click(locator=self.page_nav["block_pool_link"])
 
+    def navigate_cephblockpool(self):
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/base_ui.py:679]: On PageNavigator class, we only to "main" pages on OCP console.. Please move these methods to anther class..

File: ocs_ci/ocs/ui/validation_ui.py [hunk 2]
```diff
@@ -565,3 +512,35 @@ def check_capacity_breakdown(self, project_name, pod_name):
             logger.error(f"The pod {pod_name} not found on capacity_breakdown")
             res = False
         return res
+
+    def get_blockpools_compression_status_from_storagesystem(self) -> tuple:
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/validation_ui.py:516]: Why did you add `-> tuple`?

File: ocs_ci/ocs/ui/views.py [hunk 3]
```diff
@@ -1029,6 +1029,14 @@
         "//li[normalize-space()='StorageSystem details']",
         By.XPATH,
     ),
+    "storagesystem-details-compress-state": (
+        "#compressionStatus",
+        By.CSS_SELECTOR,
+    ),
+    "storagecluster-blockpool-details-compress-status": (
+        "article[data-test-id='compression-details-card'] dd[class='co-overview-details-card__item-value']",
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/views.py:1037]: What is an article locator?

File: ocs_ci/ocs/ui/validation_ui.py [hunk 4]
```diff
@@ -565,3 +512,35 @@ def check_capacity_breakdown(self, project_name, pod_name):
...
+            f"Get the 'Compression status' of '{constants.DEFAULT_CEPHBLOCKPOOL}'"
+        )
+        compression_status_blockpools_tab = self.get_element_text(
+            self.validation_loc["storagesystem-details-compress-state"]
+        )
+        logger.info(
+            f"Click on '{constants.DEFAULT_CEPHBLOCKPOOL}' link under BlockPools tab"
+        )
+        self.do_click(
+            self.validation_loc[f"{constants.DEFAULT_CEPHBLOCKPOOL}"],
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/validation_ui.py:540]: the type of {constants.DEFAULT_CEPHBLOCKPOOL} is set?
```
a={1}
a
{1}
type(a)
<class 'set'>```

Comment (DanielOsypenko) [reply ocs_ci/ocs/ui/validation_ui.py:540]: there are quotes and a leading "f" before quotes to use constant string var constants.DEFAULT_CEPHBLOCKPOOL

File: ocs_ci/ocs/ui/views.py (hunk 3 above)
Comment (DanielOsypenko) [reply ocs_ci/ocs/ui/views.py:1037]: it is a CSS linking article - compression-details card with its trailing value

File: ocs_ci/ocs/ui/validation_ui.py (hunk 4 above)
Comment (DanielOsypenko) [reply ocs_ci/ocs/ui/validation_ui.py:540]: yeap, your're right

File: ocs_ci/ocs/ui/validation_ui.py (hunk 2 above)
Comment (DanielOsypenko) [reply ocs_ci/ocs/ui/validation_ui.py:516]: it returns tuple compression_status_blockpools_tab, compression_status_blockpools_details

File: ocs_ci/ocs/ui/base_ui.py (hunk 1 above)
Comment (DanielOsypenko) [reply ocs_ci/ocs/ui/base_ui.py]: done -> Separated to StorageSystemNavigator

File: tests/ui/test_validation_ui.py [hunk 5]
```diff
@@ -46,6 +52,8 @@ def test_dashboard_validation_ui(self, setup_ui_class):
     @polarion_id("OCS-4642")
     @skipif_ocs_version("<4.9")
     @skipif_ui_not_support("validation")
+    @skipif_external_mode
```
Comment (am-agrawa) [inline tests/ui/test_validation_ui.py:55]: May I know why this skip was added here?

File: tests/ui/test_validation_ui.py [hunk 6]
```diff
@@ -56,3 +64,35 @@ def test_odf_storagesystems_ui(self, setup_ui_class):
         """
         validation_ui_obj = ValidationUI(setup_ui_class)
         validation_ui_obj.odf_storagesystems_ui()
+
+    @ui
+    @tier1
+    @brown_squad
```
Comment (am-agrawa) [inline tests/ui/test_validation_ui.py:70]: Should this fall under Brown squad?

File: ocs_ci/ocs/ui/validation_ui.py [hunk 7]
```diff
@@ -565,3 +512,35 @@ def check_capacity_breakdown(self, project_name, pod_name):
...
+        Initial page - Data Foundation / Storage Systems tab / StorageSystem details / ocs-storagecluster-cephblockpool
+        Get compression status from storagesystem details and ocs-storagecluster-cephblockpool
+
+        Returns:
+            tuple: String representation of 'Compression status' from StorageSystem details page and
+            String representation of 'Compression status' from ocs-storagecluster-cephblockpool page
+
+        """
+
+        self.odf_console_plugin_check()
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/validation_ui.py:527]: Why is this check removed for other functions you have created?

File: ocs_ci/ocs/ui/base_ui.py [hunk 8]
```diff
@@ -754,6 +756,92 @@ def _retrieve_current_status_from_ui():
...
+            )
+        else:
+            self.do_click(self.validation_loc["blockpools"], enable_screenshot=True)
+        self.page_has_loaded(retries=15, sleep_time=2)
+        logger.info(f"Verifying the status of '{constants.DEFAULT_CEPHBLOCKPOOL}'")
+        cephblockpool_status = self.get_element_text(
+            self.validation_loc[f"{constants.DEFAULT_CEPHBLOCKPOOL}-status"]
+        )
+        if not "Ready" == cephblockpool_status:
+            raise CephHealthException(
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/base_ui.py:839]: Should it be CephHealthException or something else?

File: ocs_ci/ocs/ui/validation_ui.py [hunk 9]
```diff
@@ -428,28 +428,8 @@ def odf_storagesystems_ui(self):
 
         """
 
-        self.odf_console_plugin_check()
-        self.navigate_odf_overview_page()
```
Comment (am-agrawa) [inline ocs_ci/ocs/ui/validation_ui.py:432]: Why is this function call removed from here?

**Review by am-agrawa — CHANGES_REQUESTED** [review]

File: ocs_ci/ocs/ui/base_ui.py [hunk 10]
```diff
@@ -383,6 +384,7 @@ def __init__(self, driver):
         self.ocp_version = get_ocp_version()
         self.ocp_version_full = version.get_semantic_ocp_version_from_config()
         self.page_nav = locators[self.ocp_version]["page"]
+        self.validation_loc = locators[self.ocp_version]["validation"]
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/base_ui.py:387]: Please do not add locators to init method.  It will fail if we use OCP4.13 and validation will not exist.

File: tests/ui/test_validation_ui.py (hunk 6 above)
Comment (ramkiperiy) [reply tests/ui/test_validation_ui.py:70]: @am-agrawa any suggestion on the squad?

Comment (am-agrawa) [reply tests/ui/test_validation_ui.py:69]: > @am-agrawa any suggestion on the squad?

It's part of dashboard validation, so I think we can keep it under Black Squad.
@ebenahar Thoughts pls?

Comment (ebenahar) [reply tests/ui/test_validation_ui.py:69]: yes, let's keep it under black squad

File: tests/ui/test_validation_ui.py [hunk 11]
```diff
@@ -46,6 +51,7 @@ def test_dashboard_validation_ui(self, setup_ui_class):
     @polarion_id("OCS-4642")
     @skipif_ocs_version("<4.9")
     @skipif_ui_not_support("validation")
+    @skipif_mcg_only
```
Comment (am-agrawa) [inline tests/ui/test_validation_ui.py:54]: I still want to understand whyare we skipping this test for mcg_only?

**Comment by am-agrawa** [comment]
Pls run the PR validation on all affected versions, i.e. 4.9, 4.10 and 4.11

**Comment by DanielOsypenko** [comment]
> Pls run the PR validation on all affected versions, i.e. 4.9, 4.10 and 4.11

Done

**Review by am-agrawa — DISMISSED** [review]

**Comment by am-agrawa** [comment]
Pls resolve conflicts.

**Review by am-agrawa — DISMISSED** [review]

**Review by OdedViner — APPROVED** [review]

File: tests/ui/test_validation_ui.py [hunk 12]
```diff
@@ -56,3 +61,35 @@ def test_odf_storagesystems_ui(self, setup_ui_class):
         """
         validation_ui_obj = ValidationUI(setup_ui_class)
         validation_ui_obj.odf_storagesystems_ui()
+
+    @ui
+    @tier1
+    @black_squad
+    @skipif_ocs_version("<4.9")
+    @skipif_external_mode
+    @skipif_mcg_only
```
Comment (ebenahar) [inline tests/ui/test_validation_ui.py:70]: polarion ID is missing

**Review by OdedViner — APPROVED** [review]

**Review by dahorak — APPROVED** [review]
